extensions/gitpod/src/remoteConnector.ts

```diff
@@ -379,64 +380,186 @@ export default class LocalApp extends Disposable {
+		const cancel = 'Cancel';
+		const action = await vscode.window.showInformationMessage('Please install "Remote - SSH" extension to connect to a Gitpod workspace.', install, cancel);
+		if (action === cancel) {
+			return false;
+		}
+
+		this.logger.info('Installing "ms-vscode-remote.remote-ssh" extension');
+
+		await vscode.commands.executeCommand('extension.open', 'ms-vscode-remote.remote-ssh');
+		await vscode.commands.executeCommand('workbench.extensions.installExtension', 'ms-vscode-remote.remote-ssh');
+
+		return true;
+	}
+
+	public async handleUri(uri: vscode.Uri) {
+		if (uri.path === RemoteConnector.AUTH_COMPLETE_PATH) {
+			this.logger.info('auth completed');
+			return;
+		}
+
+		const isRemoteSSHExtInstalled = this.ensureRemoteSSHExtInstalled();
+		if (!isRemoteSSHExtInstalled) {
+			return;
+		}
+
+		const params: SSHConnectionParams = JSON.parse(uri.query);
+		const gitpodHost = vscode.workspace.getConfiguration('gitpod').get<string>('host')!;
+		if (new URL(params.gitpodHost).host !== new URL(gitpodHost).host) {
+			const yes = 'Yes';
+			const cancel = 'Cancel';
+			const action = await vscode.window.showInformationMessage(`Trying to connect to a remote workspace in a different Gitpod Host. Continue and update 'gitpod.host' setting to '${params.gitpodHost}'?`, yes, cancel);
+			if (action === cancel) {
+				return;
+			}
+
+			await vscode.workspace.getConfiguration('gitpod').update('host', params.gitpodHost, vscode.ConfigurationTarget.Global);
+			this.logger.info(`Updated 'gitpod.host' setting to '${params.gitpodHost}' while trying to connect to a remote workspace`);
+		}
+
+		this.logger.info('Opening remote workspace', uri.toString());
+
+		let sshDestination: string | undefined;
+		try {
+			sshDestination = await this.getWorkspaceSSHDestination(params.workspaceId, params.gitpodHost);
 		} catch (e) {
-			const seeLogs = 'See Logs';
-			vscode.window.showErrorMessage(`Failed to connect to Gitpod workspace ${params.workspaceId}: ${e}`, seeLogs).then(async result => {
-				if (result !== seeLogs) {
-					return;
-				}
-				this.logger.show();
-				if (resolvedConfig) {
-					const document = await vscode.workspace.openTextDocument(vscode.Uri.file(resolvedConfig.logPath));
-					vscode.window.showTextDocument(document);
-				}
-			});
-			this.logger.error(`failed to open uri: ${e}`);
-			throw e;
+			if (e instanceof Error && e.message === 'no_running_instance') {
+				vscode.window.showErrorMessage(`Failed to connect to remote workspace: No running instance for '${params.workspaceId}'`);
+				return;
+			}
 		}
+
+		const usingSSHGateway = !!sshDestination;
+		let localAppSSHConfigPath: string | undefined;
+		if (!usingSSHGateway) {
+			vscode.window.showInformationMessage(`Falling back connecting to remote workspace using local app. You should update your Gitpod installation [enabling direct SSH access](https://github.com/gitpod-io/gitpod/blob/main/install/installer/docs/workspace-ssh-access.md)`);
```

Comment:
`showInformationMessage` => `showWarningMessage` 

`${params.host} does not support [direct SSH access](https://github.com/gitpod-io/gitpod/blob/main/install/installer/docs/workspace-ssh-access.md), connecting via the deprecated SSH tunnel over WebSocket...`